### PR TITLE
chore: fix lint errors and warnings

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Fix

Fix following `npm run lint` warnings and errors:

/Users/daviddalbusco/projects/dfinity/nns-dapp/frontend/src/lib/canisters/nns-dapp/nns-dapp.certified.idl.js
  62:9  warning  'NeuronId' is assigned a value but never used  @typescript-eslint/no-unused-vars

/Users/daviddalbusco/projects/dfinity/nns-dapp/frontend/src/lib/canisters/nns-dapp/nns-dapp.idl.js
  62:9  warning  'NeuronId' is assigned a value but never used  @typescript-eslint/no-unused-vars

/Users/daviddalbusco/projects/dfinity/nns-dapp/frontend/src/lib/derived/selected-universe.derived.ts
  47:23  warning  'featureFlags' is defined but never used  @typescript-eslint/no-unused-vars

/Users/daviddalbusco/projects/dfinity/nns-dapp/frontend/src/lib/pages/CkBTCWallet.svelte
  2:12  warning  'busy' is defined but never used  @typescript-eslint/no-unused-vars

/Users/daviddalbusco/projects/dfinity/nns-dapp/frontend/src/tests/lib/api-services/neurons.api-service.spec.ts
  173:13  warning  'identity' is defined but never used   @typescript-eslint/no-unused-vars
  174:13  warning  'certified' is defined but never used  @typescript-eslint/no-unused-vars
  218:30  warning  'certified' is defined but never used  @typescript-eslint/no-unused-vars
  322:30  warning  'certified' is defined but never used  @typescript-eslint/no-unused-vars

/Users/daviddalbusco/projects/dfinity/nns-dapp/frontend/src/tests/lib/components/project-detail/ParticipateButton.spec.ts
  62:39  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  63:54  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  64:46  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/Users/daviddalbusco/projects/dfinity/nns-dapp/frontend/src/tests/lib/services/_public/sns-proposals.services.spec.ts
  175:13  warning  'registerVoteApiSpy' is assigned a value but never used  @typescript-eslint/no-unused-vars

/Users/daviddalbusco/projects/dfinity/nns-dapp/frontend/src/tests/lib/stores/feature-flags.store.spec.ts
  80:9  error  'recordedLogs' is never reassigned. Use 'const' instead  prefer-const

/Users/daviddalbusco/projects/dfinity/nns-dapp/frontend/src/tests/lib/utils/utils.spec.ts
  19:7  warning  'callback' is defined but never used  @typescript-eslint/no-unused-vars

/Users/daviddalbusco/projects/dfinity/nns-dapp/frontend/src/tests/mocks/icrc-tokens.utils.spec.ts
  17:18  warning  '_' is defined but never used  @typescript-eslint/no-unused-vars
  26:18  warning  '_' is defined but never used  @typescript-eslint/no-unused-vars
  35:18  warning  '_' is defined but never used  @typescript-eslint/no-unused-vars

/Users/daviddalbusco/projects/dfinity/nns-dapp/frontend/src/tests/mocks/neurons.mock.ts
  29:51  error  Don't use `BigInt` as a type. Use bigint instead  @typescript-eslint/ban-types
  52:47  error  Don't use `BigInt` as a type. Use bigint instead  @typescript-eslint/ban-types
  69:52  error  Don't use `BigInt` as a type. Use bigint instead  @typescript-eslint/ban-types

### CODE REVIEW

**Review by lmuntaner — APPROVED** [review]
Thanks!

File: frontend/src/lib/derived/selected-universe.derived.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const selectedUniverseIdStore: Readable<Principal> = derived<
   Principal
 >(
   [pageUniverseIdStore, pageStore, ENABLE_CKBTC_LEDGER],
-  ([canisterId, page, featureFlags]) => {
```
Comment (dskloetd) [inline frontend/src/lib/derived/selected-universe.derived.ts:47]: featureFlags can't just be removed here but I have a fix in https://github.com/dfinity/nns-dapp/pull/1894

File: frontend/src/tests/lib/api-services/neurons.api-service.spec.ts [hunk 2]
```diff
@@ -167,19 +167,13 @@ describe("neurons api-service", () => {
     beforeEach(() => {
       jest
         .spyOn(api, "queryNeuron")
-        .mockImplementation(
-          async ({
-            neuronId,
-            identity,
-            certified,
```
Comment (dskloetd) [inline frontend/src/tests/lib/api-services/neurons.api-service.spec.ts:174]: I think it's useful to see that identity and certified are explicitly ignored here. Can we prefix them with _ instead and explicitly allow that?

I have a change to allow that here: https://github.com/dfinity/nns-dapp/compare/main...kloet/lint#diff-408265a36ffd1b4db4eb7094c64138bbdcc2cfdd1fbdb2664beba63e14accefa

File: frontend/src/tests/lib/api-services/neurons.api-service.spec.ts [hunk 3]
```diff
@@ -214,17 +208,15 @@ describe("neurons api-service", () => {
     beforeEach(() => {
       jest
         .spyOn(api, "queryNeurons")
-        .mockImplementation(
-          async ({ identity, certified }: api.ApiQueryParams) => {
```
Comment (dskloetd) [inline frontend/src/tests/lib/api-services/neurons.api-service.spec.ts:218]: same

File: frontend/src/tests/lib/api-services/neurons.api-service.spec.ts [hunk 4]
```diff
@@ -318,17 +310,15 @@ describe("neurons api-service", () => {
     beforeEach(() => {
       jest
         .spyOn(api, "queryKnownNeurons")
-        .mockImplementation(
-          async ({ identity, certified }: api.ApiQueryParams) => {
```
Comment (dskloetd) [inline frontend/src/tests/lib/api-services/neurons.api-service.spec.ts:322]: same

File: frontend/src/tests/lib/components/project-detail/ParticipateButton.spec.ts [hunk 5]
```diff
@@ -59,9 +59,19 @@ describe("ParticipateButton", () => {
     });
 
     beforeEach(() => {
-      (getOpenTicket as jest.MockedFn<any>).mockClear();
-      (initiateSnsSwapParticipation as jest.MockedFn<any>).mockClear();
-      (participateInSnsSwap as jest.MockedFn<any>).mockClear();
+      (
+        getOpenTicket as jest.MockedFn<() => Promise<SnsTicket | undefined>>
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/project-detail/ParticipateButton.spec.ts:63]: Btw, you can also just use `as jest.Mock`.
Unless you want to be able to use it as both `jest.Mock` and `() => Promise<SnsTicket | undefined>`, you don't need `jest.MockedFn<T>`.

File: frontend/src/tests/mocks/icrc-tokens.utils.spec.ts [hunk 6]
```diff
@@ -14,7 +14,7 @@ describe("icrc-tokens.utils", () => {
     it("should not return token if name is missing", () => {
       const token = mapOptionalToken(
         mockQueryTokenResponse.filter(
-          ([key, _]) => key !== IcrcMetadataResponseEntries.NAME
+          ([key]) => key !== IcrcMetadataResponseEntries.NAME
```
Comment (dskloetd) [inline frontend/src/tests/mocks/icrc-tokens.utils.spec.ts:17]: Will this work if an array of size 2 is passed in?
We can also rely on my change to allow unused _ args.

File: frontend/src/tests/mocks/icrc-tokens.utils.spec.ts [hunk 7]
```diff
@@ -23,7 +23,7 @@ describe("icrc-tokens.utils", () => {
     it("should not return token if symbol is missing", () => {
       const token = mapOptionalToken(
         mockQueryTokenResponse.filter(
-          ([key, _]) => key !== IcrcMetadataResponseEntries.SYMBOL
+          ([key]) => key !== IcrcMetadataResponseEntries.SYMBOL
```
Comment (dskloetd) [inline frontend/src/tests/mocks/icrc-tokens.utils.spec.ts:26]: same

File: frontend/src/tests/mocks/icrc-tokens.utils.spec.ts [hunk 8]
```diff
@@ -32,7 +32,7 @@ describe("icrc-tokens.utils", () => {
     it("should not return token if fee is missing", () => {
       const token = mapOptionalToken(
         mockQueryTokenResponse.filter(
-          ([key, _]) => key !== IcrcMetadataResponseEntries.FEE
+          ([key]) => key !== IcrcMetadataResponseEntries.FEE
```
Comment (dskloetd) [inline frontend/src/tests/mocks/icrc-tokens.utils.spec.ts:35]: same

File: frontend/src/tests/lib/api-services/neurons.api-service.spec.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/tests/lib/api-services/neurons.api-service.spec.ts:174]: Not against, I can live with both patterns. Ask @lmuntaner and @mstrasinskis what they think?

If they are also agree, let's do it in another PR though.

File: frontend/src/tests/mocks/icrc-tokens.utils.spec.ts (hunk 6 above)
Comment (peterpeterparker) [reply frontend/src/tests/mocks/icrc-tokens.utils.spec.ts:17]: Yep

```
const tmp = [1,2,3];
const [a] = tmp;
a === 1
```

File: frontend/src/lib/derived/selected-universe.derived.ts (hunk 1 above)
Comment (dskloetd) [reply frontend/src/lib/derived/selected-universe.derived.ts:47]: I've merged #1894. Thanks for the review.

**Review by dskloetd — APPROVED** [review]

**Pull request merged**